tests/test_report_writing.py

```diff
@@ -71,37 +86,23 @@ def test_first_filter(example_school_data):
     assert (
         (example_school_data["exit_date"].isna()) | (example_school_data["exit_date"] > "2021-7-31") | (example_school_data["exit_date"] < "2022-8-1")
     ).all()
+    pupil = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == pupil_id]
+    assert ((pupil["entry_date"] < "2022-8-1") | (pupil["entry_date"] > "2021-7-31")).all()
+    assert ((pupil["exit_date"] < "2022-8-1") | (pupil["exit_date"] > "2021-7-31") | (pupil["exit_date"].isna())).all()
```

Comment:
do these need to be ands?
```suggestion
    assert ((pupil["entry_date"] < "2022-8-1") & (pupil["entry_date"] > "2021-7-31")).all()
    assert ((pupil["exit_date"] < "2022-8-1") & (pupil["exit_date"] > "2021-7-31") | (pupil["exit_date"].isna())).all()
```

Reply:
oo yes you are right - would something like this work? because they could have either entry or exit out of the range and still be allowed in? but maybe the above solves that too!
`
assert (((pupil["entry_date"] < "2022-8-1") & (pupil["entry_date"] > "2021-7-31")) | ((pupil["exit_date"] < "2022-8-1") & (pupil["exit_date"] > "2021-7-31"))).all()`

Reply:
Ah yes it either needs to be start or end date within the window, Yeah let's go for that